src/compiler/checker.ts

```diff
@@ -14651,72 +14685,36 @@ namespace ts {
+                        if (jsdocType) {
+                            checkTypeAssignableTo(type, jsdocType, memberDecl);
+                            type = jsdocType;
                         }
                     }
-                    else if (contextualTypeHasPattern && !(getObjectFlags(contextualType) & ObjectFlags.ObjectLiteralPatternWithComputedProperties)) {
-                        // If object literal is contextually typed by the implied type of a binding pattern, and if the
-                        // binding pattern specifies a default value for the property, make the property optional.
-                        const impliedProp = getPropertyOfType(contextualType, member.escapedName);
-                        if (impliedProp) {
-                            prop.flags |= impliedProp.flags & SymbolFlags.Optional;
-                        }
 
-                        else if (!compilerOptions.suppressExcessPropertyErrors && !getIndexInfoOfType(contextualType, IndexKind.String)) {
-                            error(memberDecl.name, Diagnostics.Object_literal_may_only_specify_known_properties_and_0_does_not_exist_in_type_1,
-                                symbolToString(member), typeToString(contextualType));
-                        }
-                    }
-                    prop.declarations = member.declarations;
-                    prop.parent = member.parent;
-                    if (member.valueDeclaration) {
-                        prop.valueDeclaration = member.valueDeclaration;
+                    typeFlags |= type.flags;
+                    if (isArray(literalName)) {
+                        hasUnionedComputedProperty = true;
+                        updateIntermediateType(getUnionFromLiteralUnion(memberDecl, literalName, type));
+                        continue;
                     }
 
-                    prop.type = type;
-                    prop.target = member;
-                    member = prop;
+                    member = createProperty(memberDecl, literalName, type);
                 }
                 else if (memberDecl.kind === SyntaxKind.SpreadAssignment) {
                     if (languageVersion < ScriptTarget.ES2015) {
                         checkExternalEmitHelpers(memberDecl, ExternalEmitHelpers.Assign);
                     }
                     if (propertiesArray.length > 0) {
-                        spread = getSpreadType(spread, createObjectLiteralType(), node.symbol, propagatedFlags);
-                        propertiesArray = [];
-                        propertiesTable = createSymbolTable();
-                        hasComputedStringProperty = false;
-                        hasComputedNumberProperty = false;
-                        typeFlags = 0;
+                        updateIntermediateType(getSpreadType(intermediate, createObjectLiteralType(), node.symbol, propagatedFlags, /*fromComputedProperty*/ false));

```

Comment:
Yes, this was a mistake because it reset these properties between spreads. I thought one of the baselines improved as a result, but I don't see it now. It was something like `{ [s1]: 1, ...o, [s2]: 2, ...o }`

Reply:
computedPropertyNames5_ES5/ES6